### DISCUSSION TITLE
Javaファイルのインデント設定がファイルによって異なる場合がある

### DISCUSSION THREAD

**Discussion by rnakagawa16** [AlesInfiny/maia#1273] [05: Q&A]
# 概要

javaファイルのインデントの設定が、デフォルトでは「スペース：２」で設定されている。
しかし、いくつかのファイルにおいて、インデントの設定が「タブ：４」のように自動的に変更されてしまう場合がある。

![image](https://github.com/user-attachments/assets/818391f1-eeb4-4d18-aaa5-ec1fdb9d7be6)

# 詳細 / 機能詳細（オプション）

以下のどちらかの対処をドキュメントに示す必要がある。

①根本的な原因を特定する

②一時的な変更方法をドキュメントに示す

1) 上部ボックスに `> Indent` と入力してスペースによるインデントをクリックする
![image](https://github.com/user-attachments/assets/3ae99e39-f31c-4121-aecb-c2fc07de88a8)

2) インデントのサイズを2に変更する
![image](https://github.com/user-attachments/assets/9923a7ff-0d98-48df-af0d-8c6237f57f72)

# 完了条件

①この問題が発生する根本的な原因を特定し、対処法をドキュメントに示す
もしくは
②一時的な対処法として、変更方法をドキュメントに示す

**Comment by tsuna-can-se** [comment]
現状ではVS Codeの設定を再度行うのがもっとも単純な回避方法ですが、根本的な原因の究明には至っていません。

**Comment by kenjiyoshid-a** [comment]
こちらの記事が参考になるかもしれません。
- https://qiita.com/chihiro/items/92ab3eec1945836bd5d2

一度掲題の事象が起きた際に以下を `settings.json` に追記したところ、問題が改善されました。
```json
  "[java]": {
    "editor.detectIndentation": false
  }
```
関連箇所以外は示していません。
この事象を意図的に起こすのは難しいため、複数回再現できていません。
私がこの事象を確認できたのは、PRのコンフリクトを解消するタイミングでした。
